include/ofi_lock.h

```diff
@@ -104,21 +107,30 @@ static inline void fastlock_acquire(fastlock_t *lock)
 	assert(lock->is_initialized);
 	ret = fastlock_acquire_(&lock->impl);
 	assert(!ret);
+	lock->in_use++;
 }
 
 static inline int fastlock_tryacquire(fastlock_t *lock)
 {
+	int ret;
+
 	assert(lock->is_initialized);
-	return fastlock_tryacquire_(&lock->impl);
+	ret = fastlock_tryacquire_(&lock->impl);
+	if (!ret)
+		lock->in_use++;
+
+	return ret;
 }
 
 static inline void fastlock_release(fastlock_t *lock)
 {
 	int ret;
 
+	assert(lock->in_use);
 	assert(lock->is_initialized);
 	ret = fastlock_release_(&lock->impl);
 	assert(!ret);
+	lock->in_use--;
```

Comment:
We need to decrement in_use while still holding the lock, or the update can be lost.

Reply:
Agreed, good catch! Thanks!